// coverage-badge-creator/lib/Globals.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Globals = void 0;
const ArgvOptionsEnum_1 = require("./enums/ArgvOptionsEnum");
class Globals {
    constructor() {
        if (!Globals.instance) {
            Globals.instance = new Globals();
        }
        return Globals.instance;
    }
    static init(config) {
        this.DEFAULT_COV_PATH = (config === null || config === void 0 ? void 0 : config.coverage_file_path) || this.DEFAULT_COV_PATH;
        this.BASE_README_PATH = (config === null || config === void 0 ? void 0 : config.readmeFilePath) || this.BASE_README_PATH;
        this.BADGES = (config === null || config === void 0 ? void 0 : config.badges) || this.BADGES;
    }
    static loadArgv() {
        return new Promise((resolve, _reject) => {
            Object.keys(ArgvOptionsEnum_1.ArgvOptionsEnum).forEach((argvOptionKey) => {
                const argvOptionIndex = process.argv.indexOf(ArgvOptionsEnum_1.ArgvOptionsEnum[argvOptionKey]);
                if (argvOptionIndex !== -1) {
                    const globalKey = argvOptionKey;
                    Globals[globalKey] = process.argv[argvOptionIndex + 1];
                }
            });
            resolve();
        });
    }
}
exports.Globals = Globals;
Globals.CONFIG_PATH = './.badge-config';
Globals.DEFAULT_COV_PATH = './coverage/coverage-summary.json';
Globals.COVERAGE_CATEGORIES = ['statements', 'branches', 'functions', 'lines'];
Globals.BADGE_BASE_URL = '![](https://img.shields.io/badge/';
Globals.BADGE_BASE_URL_PATTERN = '\\!\\[]\\(https:\\/\\/img\\.shields\\.io\\/badge\\/.*prefix=&PATTERN&\\)';
Globals.BASE_README_PATH = './README.md';
Globals.BADGES = {};
