// coverage-badge-creator/lib/workers/CoverageWorker.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Coverage = void 0;
const ColorValidation_1 = require("../validations/ColorValidation");
const FileUtils_1 = require("../utils/FileUtils");
const Globals_1 = require("../Globals");
class Coverage {
    static init() {
        this.fileBody = FileUtils_1.FileUtils.readFile(Globals_1.Globals.DEFAULT_COV_PATH);
        this.CoverageJSON = FileUtils_1.FileUtils.parseFile(this.fileBody);
        this.medianCov = 0;
        return this;
    }
    static validate() {
        const coveragesCategories = {};
        Globals_1.Globals.COVERAGE_CATEGORIES.forEach((key) => {
            coveragesCategories[key] = this.getCoverageStats(this.CoverageJSON.total[key]);
            this.medianCov += this.CoverageJSON.total[key].pct;
        });
        coveragesCategories['coverage'] = this.getCoverageStats({
            pct: this.medianCov / Globals_1.Globals.COVERAGE_CATEGORIES.length,
        });
        return coveragesCategories;
    }
    static getCoverageStats(covColumn) {
        return {
            coverage: covColumn.pct,
            color: ColorValidation_1.ColorValidation.validate(covColumn.pct),
        };
    }
}
exports.Coverage = Coverage;
